src/app/activity/bounty/Paid.ts

```diff
@@ -77,7 +80,7 @@ const getDbHandler = async (request: PaidRequest): Promise<{dbBountyResult: Boun
 
 	const dbBountyResult: BountyCollection = await bountyCollection.findOne({
 		_id: new mongo.ObjectId(request.bountyId),
-		status: BountyStatus.open,
+		status: { $nin: [BountyStatus.draft, BountyStatus.deleted] }
```

Comment:
What is the purpose of this filter if we have the ID? If it fails dbBountyResult will be empty, correct? That's not handled. If it will never happen, then there's no need for this filter.

Reply:
I think we've been following a pattern that you initially set. The thinking is that there is a window of time between the validation, authorization, and activity handler, and so an extra query line saves us from making disastrous data model updates in the case of that race condition.

Reply:
OK, so then we should catch the times when a null set is returned and do something intelligent, no?

Reply:
It's a good call out. It's been in the back of my head for a while. I think the user experience we're trying to avoid is serving a runtime error message ("Sorry, something is not working. Our devs are looking into it"), vs a validation message. 

I'll table this to technical debt as well and open an issue for it - if we're going to implement null handling, we should do it across all of our activities